src/app/pages/ProposalDetailsPage/hooks.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  List,
+  useGetConsensusProposalsProposalIdVotes,
+  useGetConsensusValidators,
+} from '../../../oasis-nexus/api'
+import { Network } from '../../../types/network'
+import {
+  ExtendedVote,
+  getFilterForVoteType,
+  getRandomVote,
+  ProposalVoteValue,
+  VoteFilter,
+  VoteType,
+} from '../../../types/vote'
+import { useClientSidePagination } from '../../components/Table/useClientSidePagination'
+import { NUMBER_OF_ITEMS_ON_SEPARATE_PAGE } from '../../config'
+import { useStringInUrl } from '../../hooks/useStringInUrl'
+
+export type AllVotesData = List & {
+  isLoading: boolean
+  isError: boolean
+  loadedVotes: ExtendedVote[]
+}
+
+const DEBUG_MODE = true // TODO disable debug mode before merging
+
+const voteTable: Record<string, ProposalVoteValue> = {}
+
+const getRandomVoteFor = (address: string) => {
+  const storedVote = voteTable[address]
+  if (storedVote) return storedVote
+  const newVote = getRandomVote()
+  voteTable[address] = newVote
+  return newVote
+}
+
+const useValidatorMap = (network: Network) => {
+  const { data, isLoading, isError } = useGetConsensusValidators(network)
+  return {
+    isLoading,
+    isError,
+    map: (data as any)?.data.map ?? new Map(),
+  }
+}
+
+export const useAllVotes = (network: Network, proposalId: number): AllVotesData => {
+  const query = useGetConsensusProposalsProposalIdVotes(network, proposalId)
+  const {
+    map: validators,
+    isLoading: areValidatorsLoading,
+    isError: haveValidatorsFailed,
+  } = useValidatorMap(network)
+  const { isLoading, isError, data } = query
+
+  const extendedVotes = (data?.data.votes || []).map(
+    (vote, index): ExtendedVote => ({
+      ...vote,
+      index: index + 1,
+      areValidatorsLoading,
+      haveValidatorsFailed,
+      validator: validators.get(vote.address),
+    }),
+  )
+
+  return {
+    isLoading,
+    isError,
+    loadedVotes: DEBUG_MODE
+      ? extendedVotes.map(v => ({ ...v, vote: getRandomVoteFor(v.address) })) || []
+      : extendedVotes,
+    total_count: data?.data.total_count ?? 0,
+    is_total_count_clipped: data?.data.is_total_count_clipped ?? false,
```

Comment:
Why all the fallback values?

Reply:
Because we want to return a data structure which is a valid `List`, and in that interface, there fields are mandatory.

Keeping the `List` interface is important since we are using this data as the input of the pagination engine.
I tried relaxing the constraint, but it creates more problems than it solves.

I added a code comment to explain